sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -60,7 +61,13 @@ object JdbcUtils extends Logging {
         throw new IllegalStateException(
           s"Did not find registered driver with class $driverClass")
       }
-      driver.connect(options.url, options.asConnectionProperties)
+      val connection: Connection = driver.connect(options.url, options.asConnectionProperties)
+
+      if (connection == null) {
```

Comment:
Use `require(connection != null, ...)`. I'd also change the message and scaladoc to be a little more general, like here: `s"The driver could not open a JDBC connection. Check the URL: ${options.url}"`